src/sb-util.ts

```diff
@@ -223,22 +223,56 @@ export class Sprite extends SpriteCollection {
     }
 
     /**
-     * @return {SB3ScratchVariable}. The variables available to a Sprite include the
+     * @return {VariableCollection}. The variables available to a Sprite include the
      *      variables attached to the sprite as well as variables in the
      *      global scope, which are attached to the stage
      */
-    public variables(): SB3ScratchVariable {
+    public variables(): VariableCollection {
+        let taggedVariables: Iterable<VariableProperties>;
+
+        const stage = getSpriteMeta(this.props()).project.stage();
+        const deserializedGlobalVars: Iterable<VariableProperties> = makeIterable(
+            stage.prop(SpriteAttributes.VARIABLES),
+            deserializeVariables,
+        );
+
+        const globalVariableIter = makeIterable(
+            deserializedGlobalVars,
+            (d): Iterator<VariableProperties> =>
+                map(
+                    d,
+                    (v): VariableProperties => {
+                        getVariableMeta(v).sprite = stage.props();
+                        return v;
+                    },
+                ),
+        );
+
+        taggedVariables = globalVariableIter;
+
         if (!this.isStage()) {
             // add global scope from stage
-            const stage = getSpriteMeta(this.props()).project.stage();
-            return Object.assign(
-                {},
+            const deserializedLocalVars = makeIterable(
                 this.prop(SpriteAttributes.VARIABLES),
-                stage.prop(SpriteAttributes.VARIABLES),
+                deserializeVariables,
+            );
+
+            const localVariableIter = makeIterable(
```

Comment:
This pattern `makeIterable(iterable, iterator => map(iterator, value => ...))` has shown up a bunch. Maybe its worth having a mapIterable helper so you can skip the outer anonymous function and have `mapIterable(iterable, value => ...)`.

Or this could be a helper function exported by the meta file. Like `setMetaIterable(iterable, variable => { getVariableMeta(variable).sprite = this.props(); })`.

Reply:
`setMetaIterable` Implemented!